src/queries/fake-fei-goals.sql

```diff
@@ -19,6 +19,49 @@
 * SELECT SET_CONFIG('ssdi.regionIds','[11]',TRUE);
 */
 CREATE EXTENSION IF NOT EXISTS pg_trgm;
+------------------+------------
+WITH gnames AS (
+SELECT DISTINCT g.name gname
+FROM "Goals" g
+JOIN "Grants" gr
+  ON g."grantId" = gr.id
+WHERE g."deletedAt" IS NULL
+  AND gr.deleted = FALSE
+),
+oldest_name_instances AS (
+SELECT
+  gname,
+  MIN(id) firstgoalid
+FROM gnames
+JOIN "Goals"
+  ON gname = name
+GROUP BY 1
+),
+name_creations AS (
+SELECT
+  gname,
+  firstgoalid,
+  "createdVia" first_createdvia,
+  MIN(zag.id) firstzagid
+FROM oldest_name_instances
+JOIN "Goals" g
+  ON g.id = firstgoalid
+LEFT JOIN "ZALGoals" zag
+  ON data_id = firstgoalid
+  AND new_row_data->>'name' = gname
+GROUP BY 1,2,3
+),
+name_creators AS (
+SELECT
+  gname,
+  first_createdvia,
+  u.name creator_name
+FROM name_creations
+LEFT JOIN "ZALGoals" zag
+  ON firstzagid = zag.id
+LEFT JOIN "Users" u
+  ON zag.dml_as = u.id
+)
```

Comment:
I don't think this is a valid way to do this as most of the time the goal is the result of a copy-paste with a spelling correction. Thus the first person to do it has no correlation to others who performed the same action. This is especially notable as the same incorrect value is being used across multiple regions at roughly the same time frame. So this method of finding the first user leads to false blame. Instead lets point to who attached the goal test to the grant which is what is displayed in the UI. using something like the following:
```LEFT JOIN "GoalCollaborators" gc
ON g.id = gc."goalId"
LEFT JOIN "CollaboratorTypes" ct
ON gc."collaboratorTypeId" = ct.id
AND ct.name = 'Creator'
LEFT JOIN "ValidFor" vf
ON ct."validForId" = vf.id
AND vf.name = 'Goals'
LEFT JOIN "Users" u
ON gc."userId" = u.id```

Reply:
They decided they want a completely different thing in there anyway.